### PR TITLE
 DeckOverview screen Overflow menu icon changed #10868 error solved

### LINKED ISSUE #10868: [Suggestion] Show "Deck options" icon on Actionbar of Deck overview screen instead of its three dot menu icon
**Is your suggestion related to a problem? Please describe.**

AnkiDroid has already had the icon for "Deck options" by #10735, and there is enough space on Actionbar of Deck overview screen to add some more icons; nevertheless, the item of "Deck options" has still been imprisoned solely in the three dot menu of the screen.
image

**Describe the solution you'd like**

Show "Deck options" icon instead of the three dot menu icon.
image

**Describe alternatives you've considered**

n/a

**Additional context**

n/a

### PR DESCRIPTION
**Pull request by siddharthsingh025**
## # Error : #10868  *SOLVED ✅ * : Show "Deck options" icon on Actionbar of Deck overview screen instead of its three dot menu icon

Fixes #10868

## Output i get : 

Screenshot 2022-04-17 at 4 54 57 PM

Screenshot 2022-04-17 at 4 55 06 PM

###  **Changes i do :** 
_1. Define new style with New icon_  { https://github.com/siddharthsingh025/Anki-Android/blob/c43ae29e1ad9155e0587789534b8aa9250773b31/AnkiDroid/src/main/res/values/styles.xml#L196 }

    
    <style name="MyOverflow" parent="android:style/Widget.Holo.ActionButton.Overflow" >
    <item name="android:src">@drawable/ic_baseline_tune_24</item>
    </style>

_2.Then copy old actionbar style and ADD our new overflow style_  { https://github.com/siddharthsingh025/Anki-Android/blob/c43ae29e1ad9155e0587789534b8aa9250773b31/AnkiDroid/src/main/res/values/styles.xml#L121 }

    
    
    <style name="OverflowStyle" parent="Base.ThemeOverlay.AppCompat.ActionBar">
    
    <item name="android:textColorPrimary">@color/white</item>
    
    <item name="android:textColorSecondary">@color/white</item>
    
    <item name="popupTheme">@style/ActionBar.Popup</item>

    
    <item name="android:actionOverflowButtonStyle">@style/MyOverflow</item>

    </style>

_3.  Add our new theme {"OverflowStyle "}  to  fragment page of DeckOverview like :_  {https://github.com/siddharthsingh025/Anki-Android/blob/c43ae29e1ad9155e0587789534b8aa9250773b31/AnkiDroid/src/main/res/layout/studyoptions_fragment.xml#L5}

    android:theme="@style/OverflowStyle”   

## Checklist
_Please, go through these checks before submitting the PR._

- [ ] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [ ] You have a descriptive commit message with a short title (first line, max 50 chars).
- [ ] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [ ] You have performed a self-review of your own code
- [ ] UI changes: include screenshots of all affected screens (in particular showing any new or changed strings)
- [ ] UI Changes: You have tested your change using the [Google Accessibility Scanner](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor)

### CODE REVIEW

**Comment by david-allison** [comment]
Is there a reason this change wasn't made in the XML layout for the menu:

https://github.com/ankidroid/Anki-Android/blob/0c544371db268cb520e1f89dbc7aa2f499153452/AnkiDroid/src/main/res/menu/study_options_fragment.xml

**Comment by BrayanDSO** [comment]
We have a tune icon already `ic_tune_white.xml`, so it would be better if the icons was reused

**Comment by siddharthsingh025** [comment]
okay i will do it

**Comment by siddharthsingh025** [comment]
done : abee522062085202ee75616cc956fa5018a38c98

**Comment by siddharthsingh025** [comment]
> Is there a reason this change wasn't made in the XML layout for the menu:
> 
> https://github.com/ankidroid/Anki-Android/blob/0c544371db268cb520e1f89dbc7aa2f499153452/AnkiDroid/src/main/res/menu/study_options_fragment.xml#L22-L26


# **if you want change i can do it ??**

**Comment by david-allison** [comment]
> > Is there a reason this change wasn't made in the XML layout for the menu:
> > https://github.com/ankidroid/Anki-Android/blob/0c544371db268cb520e1f89dbc7aa2f499153452/AnkiDroid/src/main/res/menu/study_options_fragment.xml#L22-L26
> 
> # **if you want change i can do it ??**

yes please, it should make the code a lot more maintainable

**Comment by siddharthsingh025** [comment]
> yes please, it should make the code a lot more maintainable

> i think thier is no need of change in menu xml bcoz i creat theme for Overflow icon so , you want it on Deckoverview page  , its done

**Comment by siddharthsingh025** [comment]
can you help me with this failing check ?? 
 *" Unit Tests / JUnit Tests (windows-latest) (pull_request) "*

**Comment by david-allison** [comment]
> can you help me with this failing check ?? 
>  *" Unit Tests / JUnit Tests (windows-latest) (pull_request) "* 

Rebase onto main, I've fixed that in a recent commit

**Comment by siddharthsingh025** [comment]
> 

can you give the link of commit?

File: AnkiDroid/src/main/res/values/styles.xml [hunk 1]
```diff
@@ -170,4 +191,11 @@
         <!--<item name="android:textColor">?attr/md_content_color</item>-->
         <item name="android:textSize">@dimen/md_content_textsize</item>
     </style>
+
+
+    <!-- New overflow  for DeckOverview-->
+    <style name="MyOverflow" parent="android:style/Widget.Holo.ActionButton.Overflow" >
```
Comment (mikehardy) [inline AnkiDroid/src/main/res/values/styles.xml:197]: Seems reasonable - just the name doesn't mean anything without full context. Ideally the name contains the necessary context, something more like this? Though anything more descriptive would work for me

```suggestion
    <style name="TuneIconOverflow" parent="android:style/Widget.Holo.ActionButton.Overflow" >
```

(and then the reference above on line 132 would change to match)

**Review by mikehardy — CHANGES_REQUESTED** [review]

**Comment by siddharthsingh025** [comment]
can you help me with this " Emulator Tests / Android Emulator Test (pull_request) "

**Comment by mikehardy** [comment]
re-ran CI test just now, hopefully it goes green this time, might be a flaky test?

**Comment by siddharthsingh025** [comment]
what should i do now?? this PR will merged or not ??

**Review by mikehardy — APPROVED** [review]
"what you do now" is wait with just a bit of patience while all the volunteer reviewers here find another batch of time to work through the batch of PRs scanning for PRs that are ready for review again :-)

This looks good to me now, thank you!

**Comment by siddharthsingh025** [comment]
> "what you do now" is wait with just a bit of patience while all the volunteer reviewers here find another batch of time to work through the batch of PRs scanning for PRs that are ready for review again :-)
> 
> This looks good to me now, thank you!

thankyou too @mikehardy

**Review by david-allison — CHANGES_REQUESTED** [review]
Is there a reason we're not modifying a menu xml file for this?

**Comment by siddharthsingh025** [comment]
Actually @david-allison i don't understand why we need to change in  "menu / study_options_fragment"  XML file , bcoz our overflow menu icon is a part of our actionbar , and i made a style for new action bar with new overflow menu icon , and then add this to theme and after that i assign this theme to a particular layout fragment where we want change the new action bar with Tune icon menu Overflow like in studyoptions_fragment LAYOUT ........so why we need any change in menu xml of studyOption and how i can add this theme to it.

**Comment by siddharthsingh025** [comment]
> Is there a reason we're not modifying a menu xml file for this?

  Actually @david-allison i don't understand why we need to change in "menu / study_options_fragment" XML file , bcoz our       overflow menu icon is a part of our actionbar , and i made a style for new action bar with new overflow menu icon , and then    add assign this  to a theme of a particular layout fragment where we want change the new action bar with Tune icon menu Overflow like in studyoptions_fragment LAYOUT ........so why we need any change in menu xml of studyOption and how i can add this theme to it.

**Comment by siddharthsingh025** [comment]
@david-allison  are you wanting this 

https://user-images.githubusercontent.com/87073574/164956851-7e704734-fbf5-4f37-8d75-1a7c89243a07.mov


>>>>

**Comment by david-allison** [comment]
My understanding is that @snowtimeglass would prefer that the menu was removed, rather than reskinned, and the 'deck options' item appeared in the action bar instead of in the menu.

Right now, we reskin the menu item from `...` to the preference icon, but it's still a submenu with one item

**Comment by siddharthsingh025** [comment]
So , @david-allison now , can i remove the style that i creat for Overflow menu  from studyoptions_fragment LAYOUT  and than only add  Tune icon to Deckoption in "menu / study_options_fragment" XML file and make its "Showaction"  to " ifroom " .

File: AnkiDroid/src/main/res/values/styles.xml [hunk 2]
```diff
@@ -115,6 +115,10 @@
         <item name="popupTheme">@style/ActionBar.Popup</item>
     </style>
 
+
+
+
+
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/styles.xml:121]: could you revert this line change

File: AnkiDroid/src/main/res/values/styles.xml [hunk 3]
```diff
@@ -170,4 +174,5 @@
         <!--<item name="android:textColor">?attr/md_content_color</item>-->
         <item name="android:textSize">@dimen/md_content_textsize</item>
     </style>
+
```
Comment (david-allison) [inline AnkiDroid/src/main/res/values/styles.xml:177]: and this one

File: AnkiDroid/src/main/res/menu/study_options_fragment.xml [hunk 4]
```diff
@@ -1,5 +1,5 @@
 <menu xmlns:android="http://schemas.android.com/apk/res/android"
-    xmlns:ankidroid="http://schemas.android.com/apk/res-auto" >
+    xmlns:ankidroid="http://schemas.android.com/apk/res-auto">
```
Comment (david-allison) [inline AnkiDroid/src/main/res/menu/study_options_fragment.xml:2]: and this line change

**Review by david-allison — CHANGES_REQUESTED** [review]
Looks great! Just a few unnecessary line changes to clean up

File: AnkiDroid/src/main/res/menu/study_options_fragment.xml (hunk 4 above)
Comment (siddharthsingh025) [reply AnkiDroid/src/main/res/menu/study_options_fragment.xml:2]: yes

File: AnkiDroid/src/main/res/menu/study_options_fragment.xml [hunk 5]
```diff
@@ -31,7 +31,10 @@
     <!-- the title for the following option will be set dynamically -->
     <item
         android:id="@+id/action_deck_or_study_options"
-        android:title="@string/menu__deck_options"/>
+        android:title="@string/menu__deck_options"
+        android:icon="@drawable/ic_tune_white"
+        ankidroid:showAsAction="always"
+        />
```
Comment (david-allison) [inline AnkiDroid/src/main/res/menu/study_options_fragment.xml:37]: I'll approve when this is the only change in the diff.

Typically: if you avoid merge commits and rebase when bringing changes in from `main`, you can use Android Studio to compare and revert the line changes easily.

**Comment by snowtimeglass** [comment]
@siddharthsingh025 @david-allison thank you so much for tackling this issue.
 
> Yes.
> 
> My understanding is that @snowtimeglass would prefer that the menu was removed, rather than reskinned,

I'm sorry for my insufficient explanation. Either will be fine for me. I think that it's also good to leave the three dot icon as it is (if there are some appropriate items in the menu). I hope you select a not-exhausting way :-)

**Comment by david-allison** [comment]
I'm moving this to a draft to keep my queue clean, when the spacing changes are fixed, this can go in.

**Comment by david-allison** [comment]
If you click 'Files Changed', there are still files changed which shouldn't have been

**Review by david-allison — CHANGES_REQUESTED** [review]
One change in the header, then good to go, thanks!

**Comment by siddharthsingh025** [comment]
> One change in the header, then good to go, thanks!

where to change? which line?

**Review by david-allison — APPROVED** [review]
LGTM. Thanks!

**Review by mikehardy — APPROVED** [review]

**Pull request merged**